src/format/FormatChecker.c

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright (c) 2019 Takpit. All rights reserved.
+//
+
+#include "proto_nmea/format/FormatChecker.h"
+
+//region Helper Functions Declaration
+
+static int8_t validateInitialMessageFormat(const char *rawMessage);
+
+static int8_t validateMessageLengthInRange(size_t messageSize);
+
+static int8_t validateChecksumFormat(const char *message, size_t messageSize, size_t *checksumIdealIndex);
+
+static int8_t validateMessageType(const char *message);
+
+static int8_t validateChecksumData(const char *message, size_t checksumDelimiterIndex);
+
+//endregion
+
+//region API
+
+int8_t validateMessageFormat(const char *message)
+{
+    size_t messageSize = strlen(message);
+
+    return validateFixedSizeMessageFormat(message, messageSize);
+}
+
+int8_t validateFixedSizeMessageFormat(const char *rawMessage, size_t messageSize)
+{
+    int8_t errorCode = validateInitialMessageFormat(rawMessage);
+    if (errorCode)
+        return errorCode;
+
+    // Validate message ending fits protocol
+    if (rawMessage[messageSize - 2] != PROTOCOL_STOP_CHAR_1 ||
+        rawMessage[messageSize - 1] != PROTOCOL_STOP_CHAR_2)
+        return -EINVALID_PROTOCOL_ENDING;
+
+    errorCode = validateMessageLengthInRange(messageSize);
+    if (errorCode)
+        return errorCode;
+
+    size_t checksumDelimiterPosition;
+
+    errorCode = validateChecksumFormat(rawMessage, messageSize, &checksumDelimiterPosition);
+    if (errorCode)
+        return errorCode;
+
+    errorCode = validateMessageType(rawMessage);
+    if (errorCode)
+        return errorCode;
+
+    errorCode = validateChecksumData(rawMessage, checksumDelimiterPosition);
+    if (errorCode)
+        return errorCode;
+
+    return 0;
+}
+
+//endregion
+
+//region Helper Functions Definition/Implementation
+
+static int8_t validateInitialMessageFormat(const char *rawMessage)
+{
+    if (IS_NULL_STRING(rawMessage))
+        return -ENULL_STRING;
+
+    // Validate message beginning fits protocol first as it's the fastest check
+    if (*rawMessage != PROTOCOL_START_CHAR)
+        return -EINVALID_PROTOCOL_BEGINNING;
+
+    return EVALID;
+}
+
+static int8_t validateMessageLengthInRange(size_t messageSize)
+{
+    if (messageSize <= MESSAGE_MIN_LENGTH)
+    {
+        if (messageSize < MESSAGE_MIN_LENGTH)
+            return -ESHORT_MESSAGE;
+        else
+            return -EMISSING_MESSAGE_DATA;
+    }
+    else if (messageSize > MESSAGE_MAX_LENGTH)
+        return -ELONG_MESSAGE;
+
+    return EVALID;
+}
+
+static int8_t validateChecksumFormat(const char *message, size_t messageSize, size_t *checksumIdealIndex)
```

Comment:
Consider naming out-parameters with 'out' prefix to improve readability

Reply:
I disagree. The **out** modifier is mostly known to .NET developers (C# specifically), yet it's not very popular in C.